tests/models/mamba/test_modeling_mamba.py

```diff
@@ -197,7 +197,29 @@ def create_and_check_state_equivalency(self, config, input_ids, *args):
         self.parent.assertTrue(torch.allclose(torch.cat([output_one, output_two], dim=1), output_whole, atol=1e-5))
         # TODO the orignal mamba does not support decoding more than 1 token neither do we
 
-    def create_and_check_forward_and_backwards(self, config, input_ids, *args, gradient_checkpointing=False):
+    def create_and_check_mamba_cached_slow_forward_and_backwards(
+        self, config, input_ids, *args, gradient_checkpointing=False
+    ):
+        model = MambaModel(config)
+        model.to(torch_device)
+        if gradient_checkpointing:
+            model.gradient_checkpointing_enable()
+
+        # create cache
+        cache = model(input_ids, use_cache=True).cache_params
+        cache.seqlen_offset = 0
+
+        # use cache
+        token_emb = model.embeddings(input_ids)
+        outputs = model.layers[0].mixer.slow_forward(token_emb, cache)
```

Comment:
We should also have a check on the shape of `outputs` here

Reply:
Hi @amyeroberts, I've added a shape assertion for the outputs to ensure the shape is preserved (through the mixer block, i.e. token_emb.shape == outputs.shape) but I'm not sure if that's what you had in mind. If not, just let me know.

Reply:
I was thinking something like on L231, where we explicitly check for shape values against those set in the tester

Reply:
Ah ok, I see. It should be changed to that now.

Reply:
Thanks!